src/Tribe/Attendees.php

```diff
@@ -752,72 +754,97 @@ public function sanitize_csv_value( $value ) {
 	 * @since 4.6.2
 	 *
 	 */
-	public function maybe_generate_csv() {
-		if ( empty( $_GET['attendees_csv'] ) || empty( $_GET['attendees_csv_nonce'] ) || empty( $_GET['event_id'] ) ) {
+	public function maybe_generate_csv( $all = false ) {
+		if ( empty( $_GET['attendees_csv'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
 			return;
 		}
 
-		$event_id = absint( $_GET['event_id'] );
-
-		$event_id = Event::filter_event_id( $event_id, 'attendee-csv-report' );
-
-		// Verify event ID is a valid integer and the nonce is accepted.
-		if ( empty( $event_id ) || ! wp_verify_nonce( $_GET['attendees_csv_nonce'], 'attendees_csv_nonce' ) ) {
+		if ( empty( $_GET['attendees_csv_nonce'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
 			return;
 		}
 
-		$event = get_post( $event_id );
-
-		// Verify event exists and current user has access to it.
-		if (
-			! $event instanceof WP_Post
-			|| ! $this->user_can( 'edit_posts', $event_id )
-		) {
+		if ( ! current_user_can( 'manage_options' ) ) {
 			return;
 		}
 
-		// Generate filtered list of attendees.
-		$items = $this->generate_filtered_list( $event_id );
+		if ( ! $all ) {
+			if ( empty( $_GET['event_id'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+				return;
+			}
 
-		// Sanitize items for CSV usage.
-		$items = $this->sanitize_csv_rows( $items );
+			$event_id = absint( $_GET['event_id'] ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
 
-		/**
-		 * Allow for filtering and modifying the list of attendees that will be exported via CSV for a given event.
-		 *
-		 * @param array $items    The array of attendees that will be exported in this CSV file.
-		 * @param int   $event_id The ID of the event these attendees are associated with.
-		 */
-		$items = apply_filters( 'tribe_events_tickets_attendees_csv_items', $items, $event_id );
+			$event_id = Event::filter_event_id( $event_id, 'attendee-csv-report' );
+
+			// Verify event ID is a valid integer and the nonce is accepted.
+			if ( empty( $event_id ) || ! wp_verify_nonce( $_GET['attendees_csv_nonce'], 'attendees_csv_nonce' ) ) {
```

Comment:
If you move the `wp_verify_nonce` check to the top when we check if it exists, and move that check to the start, you _should_ be able to remove all the `phpcs:ignore`

Reply:
TL/DR: if the nonce is invalid we stop - immediately. We should do that before we do anything else (even checking permissions).